### ISSUE TITLE
Unable to debug azure function dotnet core project when windows user name contains spaces

### ISSUE THREAD

**Issue by DavidJFowler** [Azure/Azure-Functions#623]
I have just updated to version 15.0.31201.0 of Azure Functions and Web Jobs Tools.

I can create a new azure function project (dotnet core) with an http function. However, when I try to debug from visual studio a command window opens and then closes almost immediately.

The error message displayed is:

No executable found matching command "dotnet c:\Users\David"

Note that "David" is just the first part of my Windows login name and it contains a space.

**Comment by patricklee2** [comment]
which version of visual studios are you running?

**Comment by DavidJFowler** [comment]
Enterprise 2017 15.5.1

**Comment by tmakin** [comment]
I've run into this with VS Community 2017 15.5.1

**Comment by patricklee2** [comment]
I reproduced it in Visual Studio 2017 Enterprise 15.5.1, Azure Functions and Web Jobs Tools 15.0.31201.0
I was unable to run locally as well.
Steps:
1. Make a new user with use name "test name"
2. Install VS and Azure function tools
3. Make a new Function App and a new function
4. Debug the function locally (F5)
5. A command prompt will briefly appear with an error "dotnet C:\Users\test" and disappear

**Comment by patricklee2** [comment]
I notified the engineering team about the issue

**Comment by tmakin** [comment]
@patricklee2 Are you aware of workarounds for this? For example, a manually defined startup configuration? I'm not clear what VS is trying to do when I press "run" on a v2 function project.

**Comment by DavidJFowler** [comment]
I tried this configuration, in launchSettings.json:

```
{
  "profiles": {
    "MyApp": {
      "commandName": "Executable",
      "executablePath": "C:\\Program Files\\dotnet\\dotnet.exe",
      "commandLineArgs": "\"C:\\Users\\xxxxxx\\AppData\\Local\\Azure.Functions.V2.Cli\\2.0.1-beta\\Azure.Functions.Cli.dll\" host start \"$(TargetDir)\"",
      "workingDirectory": "$(TargetDir)"
    }
  }
}
```

but I get this error on calling the http trigger function:

```
[19/12/2017 14:55:03] A ScriptHost error has occurred
[19/12/2017 14:55:03] Exception while executing function: TestFunction. MyApp: Could not load file or assembly 'Microsoft.AspNetCore.Mvc.WebApiCompatShim, Version=2.0.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. Could not find or load a specific file. (Exception from HRESULT: 0x80131621). System.Private.CoreLib: Could not load file or assembly 'Microsoft.AspNetCore.Mvc.WebApiCompatShim, Version=2.0.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
```

**Comment by tmakin** [comment]
@DavidJFowler That sounded promising but sadly I get a different error: 

                  %%%%%%
                 %%%%%%
            @   %%%%%%    @
          @@   %%%%%%      @@
       @@@    %%%%%%%%%%%    @@@
     @@      %%%%%%%%%%        @@
       @@         %%%%       @@
         @@      %%%       @@
           @@    %%      @@
                %%
                %

    Object reference not set to an instance of an object.

Out of interest where did you find your config?

**Comment by DavidJFowler** [comment]
Educated guesswork and a bit of digging in the Azure CLI repository. [https://github.com/Azure/azure-cli](url)

**Comment by DavidJFowler** [comment]
Educated guesswork and a bit of digging in the Azure CLI [https://github.com/Azure/azure-cli](url)

**Comment by DavidJFowler** [comment]
Got this to work.

I installed the latest CLI with npm

```
npm i -g azure-functions-core-tools@core
```

and amended the command lines args accordingly:
```
"commandLineArgs": "\"C:\\Users\\xxxxxx\\AppData\\Roaming\\npm\\node_modules\\azure-functions-core-tools\\bin\\Azure.Functions.Cli.dll\" host start \"$(TargetDir)\"",
```

I also had to change the default generated function method to return IActionResult instead of HttpResponseMessage as that was causing an AutoFac exception:

```
Autofac.Core.Registration.ComponentNotRegisteredException: 'The requested service 'System.Net.Http.Formatting.IContentNegotiator' has not been registered. 
```

My function method now looks like this:

```
        [FunctionName("TestFunction")]
        public static IActionResult Run([HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = "HttpTriggerCSharp/name/{name}")]HttpRequestMessage req, string name, TraceWriter log)
        {
            log.Info("C# HTTP trigger function processed a request.");
            log.Info($"{log.GetType().FullName}");

            return new OkObjectResult($"Hello, {name}");
        }

```

**Comment by tmakin** [comment]
@DavidJFowler Nicely done, that works for me. I have found the `TargetDir` arg is redundant. So my final `launch.settings.json` looks like this:
```
{
  "profiles": {
    "Background": {
      "commandName": "Executable",
      "executablePath": "C:\\Program Files\\dotnet\\dotnet.exe",
      "commandLineArgs": "\"C:\\Users\\XXXX\\AppData\\Roaming\\npm\\node_modules\\azure-functions-core-tools\\bin\\Azure.Functions.Cli.dll\" host start",
      "workingDirectory": "$(TargetDir)"
    }
  }
}
```

**Comment by PureKrome** [comment]
Urgh - same problem here. Totally bummer that it looks like I have to npm download/install the CLI tools instead of using whatever comes with VS.

sometimes (rarely) i just want to use the stuff in VS instead of jumping into NPM cli, etc.. :(

**Comment by DavidJFowler** [comment]
Issue is still present in version 15.0.40108.0. This version does include the latest CLI so it does not need to be intalled via npm. commandLineArgs can be changed to:

"\"%LocalAppData%\\Azure.Functions.V2.Cli\\2.0.1-beta.22\\Azure.Functions.Cli.dll\" host start"

**Comment by PureKrome** [comment]
Was this supposedly fixed? link to commit ?

**Comment by tmakin** [comment]
I don't think it has been fixed at yet, as there is no mention in the release notes:
https://github.com/Azure/Azure-Functions/blob/master/VS-AzureTools-ReleaseNotes.md

**Comment by joaocpribeiro** [comment]
I created an Azure Functions project (Timer trigger) and I am having the same problem (because of my user with two names) when trying to debug. However, I do not have any launch.settings.json. I added one in the root, without success. Can anyone explain me where that file comes from?
![image](https://user-images.githubusercontent.com/26253757/35689840-b178336a-0774-11e8-9fb9-64a38be2a084.png)

**Comment by joaocpribeiro** [comment]
Ok, I found it. It should be placed in Properties folder and should be named launchSettings.json. Thanks anyway.
